panel/loading_indicator.py

```diff
@@ -0,0 +1,74 @@
+"""This module contains functionality to make any panel, i.e. layout, pane or widget, look like it's
+loading and disabled.
+
+This is very important for the user experience. A user that sees an unresponsive page without any
+indication of something happening in the background with either perceive that the page is slow or
+be in doubt if the page has stopped working.
+"""
+import panel as pn
+
+_LOADING_INDICATOR_CSS_CLASS = "panel-loading"
+
+STYLE = """
+.bk.panel-loading:before {
+position: absolute;
+height: 100%;
+width: 100%;
+content: '';
+z-index: 1000;
+background-color: rgb(255,255,255,0.50);
+border-color: lightgray;
+background-image: url('https://raw.githubusercontent.com/holoviz/panel/master/panel/assets/spinner.gif');
+background-repeat: no-repeat;
+background-position: center;
+background-size: 40px 40px;
+border-width: 1px;
+
+}
+.bk.panel-loading:hover:before {
+    cursor: progress;
+}
+"""
+pn.config.raw_css.append(STYLE)
```

Comment:
@philippjfr . I would need some guidance on where to put the style? And how to add it to Panel.